src/common/log/log.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+/**
+ * Logging utility class.
+ */
+
+import {CommandStatus} from "../commandExecutor";
+import {LogHelper, LogChannelType, LogLevel} from "./logHelper";
+import {ILogger, OutputChannelLogger, StreamLogger, ConsoleLogger} from "./loggers";
+import {OutputChannel} from "vscode";
+
+export module Log {
+    let globalLogger: ILogger;
+    /**
+     * Logs a message.
+     */
+    export function CreateGlobalLogger(targetChannel?: any) {
```

Comment:
I think we know the type of the parameter when we call this method, so instead of a switch we could have 3 methods and avoid the type checking:
createFromOutputChannel
createFromStream
createFromConsole